### PR TITLE
Scroll listener for recovery words

### PR DESCRIPTION
**Pull request by nibhar**
resolves nimiq/hub#271

### CODE REVIEW

File: src/request/export/ExportWords.js [hunk 1]
```diff
@@ -148,6 +160,22 @@ class ExportWords extends Nimiq.Observable {
         this._validateWords.setWords(words);
 
         window.location.hash = ExportWords.Pages.SHOW_WORDS;
+
+        /*
+         * scrollListener is set up here as otherwise the element is display: none;
+         * and therefor would have offsetHeight === 0
```
Comment (sisou) [inline src/request/export/ExportWords.js]:
```suggestion
         * and therefore would have offsetHeight === 0
```

File: src/request/export/ExportWords.js [hunk 2]
```diff
@@ -148,6 +160,22 @@ class ExportWords extends Nimiq.Observable {
...
+         * scrollListener is set up here as otherwise the element is display: none;
+         * and therefore would have offsetHeight === 0
+         */
+        const scrollTopThreshold = /** @type {HTMLElement} */(this.$recoveryWords.children.item(0)).offsetHeight
+            + parseInt(window.getComputedStyle(this.$recoveryWords, null).getPropertyValue('padding-top'), 10)
+            + parseInt(window.getComputedStyle(this.$recoveryWords, null).getPropertyValue('padding-bottom'), 10) / 2
+            - this.$recoveryWords.offsetHeight;
+        this.$recoveryWords.onscroll = () => {
+            if (this.$recoveryWords.scrollTop > scrollTopThreshold) {
+                console.log(this.$recoveryWords.scrollTop, scrollTopThreshold);
```
Comment (sisou) [inline src/request/export/ExportWords.js]: console.log can be removed

File: src/request/export/ExportWords.js [hunk 3]
```diff
@@ -83,11 +83,23 @@ class ExportWords extends Nimiq.Observable {
                 this._goToRecoveryWords(this._key);
             }
         });
+
         this._wordsPasswordBox.on(PasswordBox.Events.SUBMIT, this._passwordSubmitted.bind(this));
-        $recoveryWordsContinue.addEventListener('click', () => {
-            this._validateWords.reset();
-            window.location.hash = ExportWords.Pages.VALIDATE_WORDS;
+
+        this.$recoveryWordsContinue.classList.add('disabled');
```
Comment (danimoh) [inline src/request/export/ExportWords.js]: Why use a css class on the button that doesn't affect the style but only serves as a flag?
Use a boolean flag instead.

File: src/request/export/ExportWords.js [hunk 4]
```diff
@@ -148,6 +160,21 @@ class ExportWords extends Nimiq.Observable {
...
+        /*
+         * scrollListener is set up here as otherwise the element is display: none;
+         * and therefore would have offsetHeight === 0
+         */
+        const scrollTopThreshold = /** @type {HTMLElement} */(this.$recoveryWords.children.item(0)).offsetHeight
+            + parseInt(window.getComputedStyle(this.$recoveryWords, null).getPropertyValue('padding-top'), 10)
+            + parseInt(window.getComputedStyle(this.$recoveryWords, null).getPropertyValue('padding-bottom'), 10) / 2
+            - this.$recoveryWords.offsetHeight;
+        this.$recoveryWords.onscroll = () => {
+            if (this.$recoveryWords.scrollTop > scrollTopThreshold) {
```
Comment (danimoh) [inline src/request/export/ExportWords.js]: No need for complicated calulations.
Use `scollHeight` insted, e.g.: `this.$recoveryWords.scrollTop >= this.$recoveryWords.offsetHeight - x`

File: src/request/export/ExportWords.js [hunk 5]
```diff
@@ -36,6 +36,8 @@ class ExportWords extends Nimiq.Observable {
         /** @type {Key?} */
         this._key = null;
 
+        this.recoveryWordsScrolledDown = false;
```
Comment (danimoh) [inline src/request/export/ExportWords.js]: Is private and should start with an underscore.

File: src/request/export/ExportWords.js [hunk 6]
```diff
@@ -83,11 +85,21 @@ class ExportWords extends Nimiq.Observable {
...
-            this._validateWords.reset();
-            window.location.hash = ExportWords.Pages.VALIDATE_WORDS;
+
+        this.$recoveryWordsContinue.addEventListener('click', () => {
+            if (this.recoveryWordsScrolledDown) {
+                this._validateWords.reset();
+                window.location.hash = ExportWords.Pages.VALIDATE_WORDS;
+            } else {
+                this.$recoveryWords.scrollTo({
+                    top: this.$recoveryWords.scrollTop + 8, // + 1rem
```
Comment (danimoh) [inline src/request/export/ExportWords.js]: That comment is actually inaccurate. 1 rem is not always 8 px.

File: src/request/export/ExportWords.js [hunk 7]
```diff
@@ -148,6 +160,15 @@ class ExportWords extends Nimiq.Observable {
         this._validateWords.setWords(words);
 
         window.location.hash = ExportWords.Pages.SHOW_WORDS;
+
+        this.$recoveryWords.onscroll = () => {
+            if (this.$recoveryWords.scrollTop >= this.$recoveryWords.scrollHeight
+                                               - this.$recoveryWords.offsetHeight
+                                               - 16) { // half of padding-bottom: 4rem;
```
Comment (danimoh) [inline src/request/export/ExportWords.js]: I think we usually don't indent like that.

**Pull request merged**